scripts/checkpatch.pl

```diff
@@ -4419,11 +4419,13 @@ sub process {
 #  1. with a type on the left -- int [] a;
 #  2. at the beginning of a line for slice initialisers -- [0...10] = 5,
 #  3. inside a curly brace -- = { [0...10] = 5 }
+#  4. inside macro arguments, example: #define HCI_ERR(err) [err] = #err
 		while ($line =~ /(.*?\s)\[/g) {
 			my ($where, $prefix) = ($-[1], $1);
 			if ($prefix !~ /$Type\s+$/ &&
 			    ($where != 0 || $prefix !~ /^.\s+$/) &&
 			    $prefix !~ /[{,:]\s+$/ &&
+			    $prefix !~ /\#define\s+.+\s+$/ &&
```

Comment:
This change is already addressed in https://github.com/zephyrproject-rtos/zephyr/pull/73915 is it intentional to do it here again? 😃

Reply:
I added the commit to get a green CI stage. The other PR has been open for a long time already

Reply:
The other PR was merged the commit can be removed from this one

Reply:
Rebased on main now